apps/studio/src/features/editing-experience/components/utils.ts

```diff
@@ -91,3 +107,35 @@ export const generatePreviewSitemap = (
     })),
   } as IsomerGeneratedSiteProps["siteMap"]
 }
+
+export const getIframeSrc = (embedCode: string) => {
+  const elem = DOMPurify.sanitize(embedCode, {
+    ALLOWED_TAGS: ["iframe"],
+    RETURN_DOM_FRAGMENT: true,
+  })
+  const sanitizedUrl = elem.firstElementChild?.getAttribute("src")
+
+  if (sanitizedUrl === null) {
+    return undefined
+  }
+
+  return sanitizedUrl
+}
+
+export const getEmbedNameFromUrl = (url: string) =>
+  Object.entries({
+    ...MAPS_EMBED_URL_REGEXES,
+    ...VIDEO_EMBED_URL_REGEXES,
+  }).reduce<string | undefined>((acc, curr) => {
+    if (acc) {
+      // Embed name already found, return it
+      return acc
+    }
+
+    const [embedName, regex] = curr
+    if (new RegExp(regex).test(url)) {
+      return EMBED_NAME_MAPPING[embedName as keyof typeof EMBED_NAME_MAPPING]
+    }
+
+    return undefined
+  }, undefined)
```

Comment:
do we need a `reduce` here? since we aren't really needing the accumulation aspect of 

what about a simple loop with early return?

```suggestion
export const getEmbedNameFromUrl = (url: string): string | undefined => {
  const entries = Object.entries({
    ...MAPS_EMBED_URL_REGEXES,
    ...VIDEO_EMBED_URL_REGEXES,
  })

  for (const [embedName, regex] of entries) {
    if (new RegExp(regex).test(url)) {
      return EMBED_NAME_MAPPING[embedName as keyof typeof EMBED_NAME_MAPPING]
    }
  }

  return undefined
}
```

Reply:
Hmm this one was because I wanted to keep it the same as the one from `getParsedDate`:

https://github.com/opengovsg/isomer/blob/78368f5a50f98f4e62528ccc057c64fb688b2837/packages/components/src/utils/getParsedDate.ts#L13-L40

This was a comment from @seaerchin regarding using reducer pattern.

Reply:
ooo i can see the code but not the comment

Reply:
Ah, comment is here: https://github.com/opengovsg/isomer/pull/732#discussion_r1794060218